src/pymor/reductors/mt.py

```diff
@@ -0,0 +1,237 @@
+            elif which == 'LR':
+                idx = np.argsort(-poles.real)
+                dominance = np.ones(len(poles))
+            else:
+                absres = np.empty(len(poles))
+                for i in range(len(poles)):
+                    lev[i].scal(1 / lev[i].inner(fom.E.apply(rev[i]))[0][0])
+                    b_norm = fom.B.apply_adjoint(lev[i]).norm()
+                    c_norm = fom.C.apply(rev[i]).norm()
+                    absres[i] = b_norm @ c_norm
+                if which == 'NR':
+                    dominance = -(absres / np.abs(np.real(poles)))
+                    idx = np.argsort(dominance)
+
+                elif which == 'NS':
+                    dominance = -(absres / np.abs(poles))
+                    idx = np.argsort(dominance)
+
+                elif which == 'NM':
+                    idx = np.argsort(-absres)
+                    dominance = -np.array(absres)
+
+                poles = poles[idx]
+                rev = rev[idx]
+                lev = lev[idx]
+                dominance = dominance[idx]
+
+                poles = poles[:2*r]
+                lev = lev[:2*r]
+                rev = rev[:2*r]
+                dominance = dominance[:2*r]
+        elif decomposition == 'samdp':
+            poles, res, rev, lev = samdp(fom.A, fom.E,
+                                         fom.B.as_range_array(),
+                                         fom.C.as_source_array(),
+                                         r, **method_options)
+            absres = [spla.norm(res[i], ord=2) for i in range(len(poles))]
+
+            if method_options['which'] == 'LR':
+                dominance = -(absres / np.abs(np.real(poles)))
+            elif method_options['which'] == 'LS':
+                dominance = -(absres / np.abs(poles))
+            elif method_options['which'] == 'LM':
+                dominance = -np.array(absres)
+
+        if np.iscomplexobj(poles):
+            idx = sorted(range(len(poles)),
+                         key=lambda i: (dominance[i], -poles[i].real,
+                                        abs(poles[i].imag), 0 if poles[i].imag >= 0 else 1))
+            poles = poles[idx]
+            rev = rev[idx]
+            lev = lev[idx]
+
+            real_index = np.where(np.isreal(poles))[0]
+            complex_index = np.where(poles.imag > 0)[0]
+            if len(np.where(poles[:r].imag)[0]) % 2 == 1:
+                self.logger.warning('Chosen order r will split complex conjugated pair of poles.')
+
+                if (take_complex):
+                    self.logger.info("Reduced model will be complex.")
+                    poles = poles[:r]
+                    self.V = rev[:r]
+                    self.W = lev[:r]
+                else:
+                    self.logger.info("Only real part of complex conjugated pair taken.")
+                    poles = poles[:r]
+                    lev = lev[:r]
+                    rev = rev[:r]
```

Comment:
This is basically repeated in the `if` above and `else` below. It could be moved to before `if len(np.where(poles[:r].imag)[0]) % 2 == 1:`.